ports/psoc6/drivers/psoc6_gpio.c

```diff
@@ -0,0 +1,104 @@
+/* CYHAL GPIO functions */
+
+#include "psoc6_gpio.h"
+
+gpio_init_rslt gpio_init(cyhal_gpio_t pin, cyhal_gpio_direction_t direction, cyhal_gpio_drive_mode_t drive_mode, bool init_val) {
+    return cyhal_gpio_init(pin, direction, drive_mode, init_val);
+}
+
+// function to get HSIOM config of a pin
+en_hsiom_sel_t pin_get_hsiom_func(uint32_t pin) {
+    return Cy_GPIO_GetHSIOM(CYHAL_GET_PORTADDR(pin), CYHAL_GET_PIN(pin));
+}
+
+// function to get cypdl drive modes, correlated to cyhal drive modes in file: cyhal_gpio.c
+uint32_t gpio_get_cypdl_drive(uint32_t pin) {
```

Comment:
gpio_get_drive could be better or?

Reply:
In any case, we should try to stay at HAL level.

Reply:
> In any case, we should try to stay at HAL level.

The reason to go below the HAL into the PDL was that, not everything is implemented in the HAL. Sometimes an access to the PDL might be needed on a case basis. In this case, we are trying to map params from the MPY side (direction,drive and pull) to the param on the CYHAL/PDL side(drive). This needs some additional infrastructure which are not/might not be present in the HAL at the moment fully. 

After discussions we agreed that PDL has to be used with caution since its much closer to the HW layer and necessary checks must be implemented by us.